src/goto-instrument/accelerate/scratch_program.h

```diff
@@ -43,12 +43,15 @@ class scratch_programt:public goto_programt
       ns(symbol_table, symex_symbol_table),
       equation(),
       path_storage(),
-      symex(mh, symbol_table, equation, path_storage),
+      options(),
+      symex(mh, symbol_table, equation, options, path_storage),
       satcheck(util_make_unique<satcheckt>()),
       satchecker(ns, *satcheck),
       z3(ns, "accelerate", "", "", smt2_dect::solvert::Z3),
       checker(&z3) // checker(&satchecker)
   {
+    options.set_option("simplify", true);
```

Comment:
Comment *why* these are being unconditionally set, regardless of the user's selections?

Reply:
Also suggest commenting that these are local alterations (only applicable to this program), since this sort of code in cbmc is often making global option changes.

Reply:
@smowton I don't know much about the acceleration code, so I have no idea why these are always set...perhaps @tautschnig knows, or knows who knows? I guess this is Matt Lewis' code, but the migration to GitHub happened after Matt left so I can't ping him on here

Reply:
That's just me wanting to preserve the prior behaviour, where `goto_symext` would unconditionally set them, which it now no longer does (and therefore they have to be set here). That said, running without simplification turned on would likely yield poor performance. The `"assertions"` option could be scratched, it has no effect here (or in `goto_symext`, it only matters for `goto_check`).

Reply:
With the above said, @karkhaz could you change this to only set the `"simplify"` option, and add a comment that that's for performance reasons?

Reply:
@tautschnig will do so. However, unless I'm mistaken, the `goto_symext` constructed by `bmct` does +not+ get the `"simplify"` option switched on (or the `"assertions"` option). Does this also matter for performance, or does it really only matter for the acceleration code?

If it does matter for `bmct`, then I will have to make `bmct`'s option reference non-const, so that I can set `"simplify"` to `true`. I would do this in the `bmct::setup()` method, I suppose. Please let me know if I should do this.

Reply:
I'd say cbmc_parse_optionst::set_default_options does take care of those?

Reply:
Amazing that I didn't realise that, given that I wrote that method. Thanks.